compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUVectorLowering.cpp

```diff
@@ -74,6 +74,17 @@ void LLVMCPUVectorLoweringPass::runOnOperation() {
     llvm::dbgs() << "\n\n";
   });
 
+  // TODO(dcaballe): We should split this pass into smaller/simpler ones and
+  // replace the code below with a full canonicalize pass.
+  {
+    RewritePatternSet patterns(ctx);
+    for (auto *dialect : ctx->getLoadedDialects())
+      dialect->getCanonicalizationPatterns(patterns);
+    for (RegisteredOperationName op : ctx->getRegisteredOperations())
+      op.getCanonicalizationPatterns(patterns, ctx);
+    (void)applyPatternsAndFoldGreedily(funcOp, std::move(patterns));
+  }
```

Comment:
Do we need all the canonicalization patterns? Are canonicalization patterns of vector dialect and memref dialect enough?